src/aiida/transports/plugins/ssh_async.py

```diff
@@ -572,7 +572,7 @@ async def puttree_async(
             raise OSError('Cannot copy a directory into a file')
 
         if not await self.isdir_async(remotepath):  # in this case copy things in the remotepath directly
-            await self.mkdir_async(remotepath)  # and make a directory at its place
+            await self.makedirs_async(remotepath, ignore_existing=overwrite)  # and make a directory at its place
```

Comment:
```suggestion
            await self.makedirs_async(remotepath)  # and make a directory at its place
```
`remotepath` will not ever exist in this case, since the `if` condition is already checking that :)

Reply:
Fair point! Will adapt :)